### ISSUE TITLE
TCH005 autofix deletes else block, removes functional code.

### ISSUE THREAD

**Issue by Skylion007** [astral-sh/ruff#11368]
I applied TCH autofixes to pytorch recently, and hit a pretty gnarly false positive here.
In this case `TCH005` TYPE_CHECKING "pass" is used to trigger the else block. When the autofix is applied, instead of doing something sensible like changing it to an `if not TYPE_CHECKING` it just deletes the entire block. Here is the commit ID that reverted it:
https://github.com/pytorch/pytorch/pull/125536/commits/b7f8f56a079399a32129e3fcc424ae88202baea5#diff-5f3d4caa0693a716fc46fd7f6339312f1b5f0bf89e3a3ff58e9dc13a9486b17aR1103

```python
if TYPE_CHECKING:
   pass
else:
   print("IMPORTANT PRINT STATEMENT HERE")
```
the fixit would delete the entire code block instead of doing a more sensible fix.
